jekyll/_cci2/gh-bb-integration.md

```diff
@@ -286,5 +291,24 @@ jobs:
             - "SO:ME:FIN:G:ER:PR:IN:T"
 ```
 
-When you push to your Bitbucket project from a job,
-CircleCI will use the SSH key you added.
+### How are these keys used?
+
+When we build your project, we install the private key into the .ssh directory, and configure ssh to use it when communicating with your version control provider. Therefore, it gets used for:
+
+- checking out the main project
+- checking out any GitHub-hosted submodules
+- checking out any GitHub-hosted private dependencies
+- automatic git merging/tagging/etc.
+
+That's why a deploy key isn't sufficiently powerful for projects with additional private dependencies!
+
+### What about security?
+
+The private keys of the checkout keypairs we generate never leave our systems
+(only the public key is transmitted to GitHub) and are safely encrypted in storage. However, since they are installed into your build containers, any code that you run in CircleCI can read them. You shouldn't push untrusted code to CircleCI!
+
+**Isn't there a middle ground between deploy keys and user keys?**
+
+Not really (unfortunately!). Deploy keys and user keys are the only key types that GitHub supports. Deploy keys are globally unique (i.e. there's no way to make a deploy key with access to multiple repositories) and user keys have no notion of _scope_ separate from the user they're associated with.
```

Comment:
Remove first sentence fragment
i.e.-> for example
there's-> there is or 'no mechanism exists to'
they're-> user associated with them